### ISSUE TITLE
Thread 'tokio-runtime-worker' panicked regularly

### ISSUE THREAD

**Issue by mikespook** [paritytech/polkadot-sdk#5133]
Hi there,

This is Mike from Onfinality.io. We're currently using the `parity/polkadot-parachain:1.14.0` Docker image to run several chains in our Kubernetes environment, including:

- Polkadot (5 nodes across 2 clusters)
- Kusama (5 nodes across 2 clusters)
- Statemint (2 nodes across 2 clusters)
- Bridge-hub-polkadot (2 nodes across 2 clusters)
- Collectives-polkadot (1 node in a cluster)
- Westend (3 nodes in a cluster)

We're encountering an issue across all 18 nodes, where they experience a panic after running for a while. The stack trace information indicates that the nodes operate normally until a panic is caught by the handler. It seems like there may be a race condition occurring. 

Could you please take a look and let us know if you need any additional information? Thank you!

```
chain-node 2024-07-24 22:45:12 [Parachain] 💤 Idle (29 peers), best: #3077297 (0xd548…c163), finalized #3077295 (0x64de…8843), ⬇ 6.8kiB/s ⬆ 2.5kiB/s    
chain-node 2024-07-24 22:45:12 [Relaychain] 🏆 Imported #21792044 (0x8691…423a → 0x6050…5cea)    
chain-node 2024-07-24 22:45:12 [Relaychain] 💤 Idle (29 peers), best: #21792044 (0x6050…5cea), finalized #21792041 (0x76b5…c62f), ⬇ 223.9kiB/s ⬆ 199.6kiB/s    
chain-node 2024-07-24 22:45:17 [Parachain] 💤 Idle (29 peers), best: #3077297 (0xd548…c163), finalized #3077295 (0x64de…8843), ⬇ 0.9kiB/s ⬆ 1.2kiB/s    
chain-node 2024-07-24 22:45:17 [Relaychain] 💤 Idle (30 peers), best: #21792044 (0x6050…5cea), finalized #21792042 (0xa190…6c03), ⬇ 254.4kiB/s ⬆ 230.7kiB/s    
chain-node 2024-07-24 22:45:18 [Relaychain] 🏆 Imported #21792045 (0x6050…5cea → 0x68bb…60c8)    
chain-node 2024-07-24 22:45:18 [Relaychain] 🆕 Imported #21792045 (0x6050…5cea → 0x9b58…ae19)    
chain-node 2024-07-24 22:45:19 [Parachain] 🏆 Imported #3077298 (0xd548…c163 → 0xe16d…8640)    
chain-node 2024-07-24 22:45:19 [Parachain] 🆕 Imported #3077298 (0xd548…c163 → 0x79b3…49d7)    
chain-node 2024-07-24 22:45:22 [Parachain] 💤 Idle (29 peers), best: #3077298 (0xe16d…8640), finalized #3077295 (0x64de…8843), ⬇ 71.2kiB/s ⬆ 21.3kiB/s    
chain-node 2024-07-24 22:45:22 [Relaychain] 💤 Idle (30 peers), best: #21792045 (0x68bb…60c8), finalized #21792042 (0xa190…6c03), ⬇ 240.4kiB/s ⬆ 365.6kiB/s    
chain-node 2024-07-24 22:45:26 [Relaychain] 🏆 Imported #21792046 (0x68bb…60c8 → 0x26d7…c846)    
chain-node 2024-07-24 22:45:27 [Parachain] 💤 Idle (29 peers), best: #3077298 (0xe16d…8640), finalized #3077296 (0x88d2…4a95), ⬇ 0.6kiB/s ⬆ 0.9kiB/s    
chain-node 2024-07-24 22:45:27 [Relaychain] 💤 Idle (29 peers), best: #21792046 (0x26d7…c846), finalized #21792043 (0x8691…423a), ⬇ 159.2kiB/s ⬆ 240.3kiB/s    
chain-node 2024-07-24 22:45:30 [Relaychain] 🏆 Imported #21792047 (0x26d7…c846 → 0x51a3…5790)    
chain-node 2024-07-24 22:45:30 [Parachain] ♻️  Reorg on #3077298,0xe16d…8640 to #3077298,0x79b3…49d7, common ancestor #3077297,0xd548…c163    
chain-node 2024-07-24 22:45:30 [Parachain] 🏆 Imported #3077299 (0x79b3…49d7 → 0x3338…cec1)    
chain-node 2024-07-24 22:45:32 [Parachain] 💤 Idle (29 peers), best: #3077299 (0x3338…cec1), finalized #3077296 (0x88d2…4a95), ⬇ 41.8kiB/s ⬆ 13.0kiB/s    
chain-node 2024-07-24 22:45:32 [Relaychain] 💤 Idle (29 peers), best: #21792047 (0x51a3…5790), finalized #21792043 (0x8691…423a), ⬇ 458.1kiB/s ⬆ 269.2kiB/s    
chain-node 2024-07-24 22:45:36 [Relaychain] 🏆 Imported #21792048 (0x51a3…5790 → 0x6d04…c4eb)    
chain-node 2024-07-24 22:45:36 [Relaychain] 🆕 Imported #21792048 (0x51a3…5790 → 0x8a61…07e1)    
chain-node 2024-07-24 22:45:37 [Parachain] 💤 Idle (29 peers), best: #3077299 (0x3338…cec1), finalized #3077297 (0xd548…c163), ⬇ 1.7kiB/s ⬆ 2.2kiB/s    
chain-node 2024-07-24 22:45:37 [Relaychain] 💤 Idle (29 peers), best: #21792048 (0x6d04…c4eb), finalized #21792045 (0x68bb…60c8), ⬇ 381.8kiB/s ⬆ 192.6kiB/s    
chain-node 2024-07-24 22:45:42 [Parachain] 💤 Idle (29 peers), best: #3077299 (0x3338…cec1), finalized #3077297 (0xd548…c163), ⬇ 0.6kiB/s ⬆ 0.6kiB/s    
chain-node 2024-07-24 22:45:42 [Relaychain] 🏆 Imported #21792049 (0x6d04…c4eb → 0x786f…a68c)    
chain-node 2024-07-24 22:45:42 [Relaychain] ♻️  Reorg on #21792049,0x786f…a68c to #21792049,0xf0b9…6222, common ancestor #21792048,0x6d04…c4eb    
chain-node 2024-07-24 22:45:42 [Relaychain] 🏆 Imported #21792049 (0x6d04…c4eb → 0xf0b9…6222)    
chain-node 2024-07-24 22:45:42 [Parachain] 🏆 Imported #3077300 (0x3338…cec1 → 0x778c…8840)    
chain-node 2024-07-24 22:45:42 [Relaychain] 💤 Idle (29 peers), best: #21792049 (0xf0b9…6222), finalized #21792046 (0x26d7…c846), ⬇ 269.6kiB/s ⬆ 311.0kiB/s    
chain-node 2024-07-24 22:45:42 [Parachain] 🆕 Imported #3077300 (0x3338…cec1 → 0x51d6…a010)    
chain-node 2024-07-24 22:45:47 [Parachain] 💤 Idle (29 peers), best: #3077300 (0x778c…8840), finalized #3077297 (0xd548…c163), ⬇ 67.5kiB/s ⬆ 15.3kiB/s    
chain-node 2024-07-24 22:45:47 [Relaychain] 💤 Idle (29 peers), best: #21792049 (0xf0b9…6222), finalized #21792046 (0x26d7…c846), ⬇ 290.2kiB/s ⬆ 269.3kiB/s    
chain-node 2024-07-24 22:45:48 [Relaychain] 🏆 Imported #21792050 (0xf0b9…6222 → 0x27d2…f1a3)    
chain-node 2024-07-24 22:45:52 [Parachain] 💤 Idle (29 peers), best: #3077300 (0x778c…8840), finalized #3077297 (0xd548…c163), ⬇ 1.2kiB/s ⬆ 1.1kiB/s    
chain-node 2024-07-24 22:45:52 [Relaychain] 💤 Idle (29 peers), best: #21792050 (0x27d2…f1a3), finalized #21792046 (0x26d7…c846), ⬇ 236.2kiB/s ⬆ 269.7kiB/s    
chain-node 2024-07-24 22:45:54 [Relaychain] 🏆 Imported #21792051 (0x27d2…f1a3 → 0xe9bc…0dfc)    
chain-node 2024-07-24 22:45:54 [Parachain] 🏆 Imported #3077301 (0x778c…8840 → 0xfaf1…bfbb)    
chain-node 2024-07-24 22:45:57 [Parachain] 💤 Idle (29 peers), best: #3077301 (0xfaf1…bfbb), finalized #3077298 (0x79b3…49d7), ⬇ 41.4kiB/s ⬆ 12.5kiB/s    
chain-node 2024-07-24 22:45:57 [Relaychain] 💤 Idle (29 peers), best: #21792051 (0xe9bc…0dfc), finalized #21792048 (0x6d04…c4eb), ⬇ 355.2kiB/s ⬆ 378.1kiB/s    
chain-node 2024-07-24 22:46:00 [Relaychain] 🏆 Imported #21792052 (0xe9bc…0dfc → 0xe7c7…9097)    
chain-node 2024-07-24 22:46:00 [Relaychain] ♻️  Reorg on #21792052,0xe7c7…9097 to #21792052,0xd9c0…9c87, common ancestor #21792051,0xe9bc…0dfc    
chain-node 2024-07-24 22:46:00 [Relaychain] 🏆 Imported #21792052 (0xe9bc…0dfc → 0xd9c0…9c87)    
chain-node 2024-07-24 22:46:02 [Parachain] 💤 Idle (29 peers), best: #3077301 (0xfaf1…bfbb), finalized #3077298 (0x79b3…49d7), ⬇ 1.6kiB/s ⬆ 2.8kiB/s    
chain-node 2024-07-24 22:46:02 [Relaychain] 💤 Idle (29 peers), best: #21792052 (0xd9c0…9c87), finalized #21792048 (0x6d04…c4eb), ⬇ 311.8kiB/s ⬆ 240.8kiB/s    
chain-node 2024-07-24 22:46:06 [Relaychain] 🏆 Imported #21792053 (0xd9c0…9c87 → 0x1c5f…388b)    
chain-node 2024-07-24 22:46:06 [Relaychain] ♻️  Reorg on #21792053,0x1c5f…388b to #21792053,0x5dce…4c62, common ancestor #21792052,0xd9c0…9c87    
chain-node 2024-07-24 22:46:06 [Relaychain] 🏆 Imported #21792053 (0xd9c0…9c87 → 0x5dce…4c62)    
chain-node 2024-07-24 22:46:06 [Parachain] 🏆 Imported #3077302 (0xfaf1…bfbb → 0x533e…37ff)    
chain-node 2024-07-24 22:46:06 [Parachain] 🆕 Imported #3077302 (0xfaf1…bfbb → 0x1230…bfd0)    
chain-node 2024-07-24 22:46:07 [Parachain] 💤 Idle (29 peers), best: #3077302 (0x533e…37ff), finalized #3077299 (0x3338…cec1), ⬇ 48.1kiB/s ⬆ 10.3kiB/s    
chain-node 2024-07-24 22:46:07 [Relaychain] 💤 Idle (29 peers), best: #21792053 (0x5dce…4c62), finalized #21792049 (0xf0b9…6222), ⬇ 364.3kiB/s ⬆ 358.1kiB/s    
chain-node 2024-07-24 22:46:12 [Parachain] 💤 Idle (29 peers), best: #3077302 (0x533e…37ff), finalized #3077299 (0x3338…cec1), ⬇ 7.9kiB/s ⬆ 8.2kiB/s    
chain-node 2024-07-24 22:46:12 [Relaychain] 🏆 Imported #21792054 (0x5dce…4c62 → 0x926d…7a6b)    
chain-node 2024-07-24 22:46:12 [Relaychain] 💤 Idle (29 peers), best: #21792054 (0x926d…7a6b), finalized #21792049 (0xf0b9…6222), ⬇ 320.7kiB/s ⬆ 193.0kiB/s    
chain-node 2024-07-24 22:46:17 [Parachain] 💤 Idle (29 peers), best: #3077302 (0x533e…37ff), finalized #3077300 (0x778c…8840), ⬇ 1.5kiB/s ⬆ 1.4kiB/s    
chain-node 2024-07-24 22:46:17 [Relaychain] 💤 Idle (29 peers), best: #21792054 (0x926d…7a6b), finalized #21792051 (0xe9bc…0dfc), ⬇ 256.0kiB/s ⬆ 291.1kiB/s    
chain-node 2024-07-24 22:46:18 [Parachain] ♻️  Reorg on #3077302,0x533e…37ff to #3077303,0xa581…4963, common ancestor #3077301,0xfaf1…bfbb    
chain-node 2024-07-24 22:46:18 [Parachain] 🏆 Imported #3077303 (0x1230…bfd0 → 0xa581…4963)    
chain-node 2024-07-24 22:46:18 [Relaychain] 🏆 Imported #21792055 (0x926d…7a6b → 0xa75e…a448)    
chain-node 2024-07-24 22:46:22 [Parachain] 💤 Idle (29 peers), best: #3077303 (0xa581…4963), finalized #3077301 (0xfaf1…bfbb), ⬇ 10.2kiB/s ⬆ 84.9kiB/s    
chain-node 2024-07-24 22:46:22 [Relaychain] 💤 Idle (29 peers), best: #21792055 (0xa75e…a448), finalized #21792053 (0x5dce…4c62), ⬇ 245.4kiB/s ⬆ 252.9kiB/s    
chain-node 
chain-node ====================
chain-node 
chain-node Version: 1.14.0-364961445b7
chain-node 
chain-node    0: sp_panic_handler::set::{{closure}}
chain-node    1: std::panicking::rust_panic_with_hook
chain-node    2: std::panicking::begin_panic::{{closure}}
chain-node    3: std::sys_common::backtrace::__rust_end_short_backtrace
chain-node    4: std::panicking::begin_panic
chain-node    5: <quicksink::SinkImpl<S,F,T,A,E> as futures_sink::Sink<A>>::poll_ready
chain-node    6: futures_util::future::either::if_std::<impl futures_io::if_std::AsyncWrite for futures_util::future::either::Either<A,B>>::poll_write
chain-node    7: <libp2p_noise::io::framed::NoiseFramed<T,S> as futures_sink::Sink<&alloc::vec::Vec<u8>>>::poll_ready
chain-node    8: <libp2p_noise::io::Output<T> as futures_io::if_std::AsyncWrite>::poll_write
chain-node    9: <yamux::frame::io::Io<T> as futures_sink::Sink<yamux::frame::Frame<()>>>::poll_ready
chain-node   10: yamux::connection::Connection<T>::poll_next_inbound
chain-node   11: <libp2p_core::muxing::boxed::Wrap<T> as libp2p_core::muxing::StreamMuxer>::poll
chain-node   12: <libp2p_core::muxing::boxed::Wrap<T> as libp2p_core::muxing::StreamMuxer>::poll
chain-node   13: libp2p_swarm::connection::Connection<THandler>::poll
chain-node   14: libp2p_swarm::connection::pool::task::new_for_established_connection::{{closure}}
chain-node   15: <futures_util::future::select::Select<A,B> as core::future::future::Future>::poll
chain-node   16: <tracing_futures::Instrumented<T> as core::future::future::Future>::poll
chain-node   17: tokio::runtime::task::raw::poll
chain-node   18: tokio::runtime::scheduler::multi_thread::worker::Context::run_task
chain-node   19: tokio::runtime::scheduler::multi_thread::worker::run
chain-node   20: tokio::runtime::task::raw::poll
chain-node   21: std::sys_common::backtrace::__rust_begin_short_backtrace
chain-node   22: core::ops::function::FnOnce::call_once{{vtable.shim}}
chain-node   23: std::sys::pal::unix::thread::Thread::new::thread_start
chain-node   24: <unknown>
chain-node   25: <unknown>
chain-node 
chain-node 
chain-node Thread 'tokio-runtime-worker' panicked at 'SinkImpl::poll_ready called after error.', /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/quicksink-0.1.2/src/lib.rs:158
chain-node 
chain-node This is a bug. Please report it at:
chain-node 
chain-node     https://github.com/paritytech/polkadot-sdk/issues/new
chain-node 
Stream closed EOF for ws-6752764341753147392/node-7185747462404251648-0 (chain-node)
```

![Screenshot 2024-07-24 at 16 39 06](https://github.com/user-attachments/assets/d3bbc808-9e5f-4fb2-bacf-dcc0857df4c7)

**Comment by alindima** [comment]
Looks like the same issue as in https://github.com/paritytech/polkadot-sdk/issues/4934.
I think it was fixed in master but not yet released. CC @lexnv.

**Comment by bkchr** [comment]
Yeah it is a duplicate of https://github.com/paritytech/polkadot-sdk/issues/4934. It will be fixed with 1.15

**Comment by lexnv** [comment]
@mikespook we have released 1.14.1 as well as updated the docker images. Feel free to update anytime! 

This is all thanks to an amazing job done by @EgorPopelyaev 🙏

**Comment by mikespook** [comment]
Thanks! Appreciate! I have asked the team to update it ASAP.
